Readme.md

```diff
@@ -222,60 +222,113 @@ if(maybeABasket.isNothing()) {
 ## Maybe api
 
 - Maybe take a value and return a maybe, if the passed value is null or undefined, all operation on the maybe to change the value will be ignored.
+
 - Maybe.of: same as Maybe
-- Maybe.lift: Take a function and wrap its return value in Maybe
+
+- Maybe.lift: Takes a function and wrap its return value in Maybe
 ```ts
-lift<A, B>(fn: (v: A) => B): (v: A): Maybe<B>;
+// Maybe.lift(fn: A => B): (v: A) => Maybe<B>;
+const fn = Maybe.lift(v => v * 2);
+// is the same as
+const fn = v => Maybe.of(v * 2);
 ```
+
 - maybe.map: Take a function and apply it to the value if there is one
 ```ts
-map<A, B>(this: Maybe<A>, fn: (value: A) => B): Maybe<B>;
-map<A, B>(this: Maybe<nothing>, fn: (value: A) => B): Maybe<nothing>;
+// Maybe<A>.map<fn: A => B>: Maybe<B>;
+Maybe.of(5).map(v => v * 2);
+-> Maybe<10>;
+// Maybe<null | undefined>.map(fn: A => B): Maybe<null | undefined>;
+Maybe.of(null).map(v => v * 2);
+-> Maybe<null>;
 ```
 - maybe.flatten: If the value is another Maybe merge the two together. If there is no value, do nothing.
```

Comment:
merge the current maybe with the one given in argument if there is one. Otherwise, does nothing.